### PR TITLE
chore(test_visibility): rename and privatize CI Visibility API to Test Visibility

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
This PR renames the unreleased, to-be-externally-facing Test Visibility-related methods/classes/etc to use some variations of "test visibility"-themed names to reflect the fact that the CI Visibility product is now known as Test Visibility.

It also takes several pieces that were previously public and creates a new, internal, test visibility API that extends the external one. 

Notably:
* New core event handlers for enabling/disabling/getting status of test visibility added to the tracer's global handlers (test visibility-related handlers continue to only be added if the `CIVisibility` service is imported) 
* Externally facing changes
  * event hub names are renamed from `ci_visibility.*` to `test_visibility.*`
  * `CI*` API classes are renamed to `Test*`
  * Move of `*ItemId` classes to a separate file to reduce circular dependencies between external and internal API
  * Moved to internal interface:
    * ITR mixin classes
    * Methods to access item spans directly 
* Internal changes
  * `CIVisibility*` API classes renamed to `TestVisibility`
* No test definition changes except for snapshot data
* Unrelated bundled changes
  * Whether to use test-level or suite-level skipping in ITR is now configured in the introduced `test_visibility` integration config (renamed from `ci_visibility`)
  * the v2 pytest plugin now uses this configuration item to enable suite-level skipping

One important note is that `DD_CIVISIBILITY_*` environment variables remain untouched. This does lead to the incongruity of configuring Test Visibility with CI Visibility-named vars.

There are no release notes because the public API components are unreleased and marked as subject to change in comments.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/ext/test_visibility/item_ids.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+"""Provides identifier classes for items used in the Test Visibility API"""
```
Comment (romainkomorndatadog) [inline ddtrace/ext/test_visibility/item_ids.py:1]:
```suggestion
"""Provides identifier classes for items used in the Test Visibility API

NOTE: BETA - this API is currently in development and is subject to change.
"""
```

File: ddtrace/_trace/trace_handlers.py [hunk 2]
```diff
@@ -802,6 +820,10 @@ def listen():
     core.on("redis.async_command.post", _on_redis_command_post)
     core.on("redis.command.post", _on_redis_command_post)
 
+    core.on("test_visibility.enable", _on_test_visibility_enable)
+    core.on("test_visibility.disable", _on_test_visibility_disable)
+    core.on("test_visibility.is_enabled", _on_test_visibility_is_enabled, "is_enabled")
```
Comment (romainkomorndatadog) [inline ddtrace/_trace/trace_handlers.py:825]: This wants review from @DataDog/apm-core-python .

**Review by vitor-de-araujo — APPROVED** [review]

**Review by juan-fernandez — APPROVED** [review]

File: tests/ci_visibility/api/fake_runner_all_fail.py [hunk 3]
```diff
@@ -4,123 +4,124 @@
 from pathlib import Path
 import sys
 
-from ddtrace.ext.ci_visibility import api
+from ddtrace.ext.test_visibility import api
+import ddtrace.ext.test_visibility.item_ids
```
Comment (erikayasuda) [inline tests/ci_visibility/api/fake_runner_all_fail.py:8]: nit: Should we alias this to be shorter?

File: tests/ci_visibility/api/fake_runner_all_itr_skip_suite_level.py [hunk 4]
```diff
@@ -4,116 +4,117 @@
 from pathlib import Path
 from unittest import mock
 
-from ddtrace.ext.ci_visibility import api
+from ddtrace.ext.test_visibility import api
+import ddtrace.ext.test_visibility.item_ids
```
Comment (erikayasuda) [inline tests/ci_visibility/api/fake_runner_all_itr_skip_suite_level.py:8]: nit: Same comment here, shorter alias maybe?

File: ddtrace/ext/test_visibility/item_ids.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+"""Provides identifier classes for items used in the Test Visibility API
+
+NOTE: BETA - this API is currently in development and is subject to change.
```
Comment (erikayasuda) [inline ddtrace/ext/test_visibility/item_ids.py:3]: Maybe also include this in a release note as a new feature?

File: tests/ci_visibility/api/fake_runner_all_fail.py (hunk 3 above)
Comment (erikayasuda) [reply tests/ci_visibility/api/fake_runner_all_fail.py:8]: I guess it's just a test file, so it would really just be for internal readability 😅

**Review by erikayasuda — APPROVED** [review]

File: ddtrace/ext/test_visibility/item_ids.py (hunk 5 above)
Comment (romainkomorndatadog) [reply ddtrace/ext/test_visibility/item_ids.py:3]: I'm working on a PR for documentation, and then we'll put a release note.

File: tests/ci_visibility/api/fake_runner_all_itr_skip_suite_level.py (hunk 4 above)
Comment (romainkomorndatadog) [reply tests/ci_visibility/api/fake_runner_all_itr_skip_suite_level.py:8]: Maybe in a different PR. 😓

File: ddtrace/internal/ci_visibility/api/_coverage_data.py [hunk 6]
```diff
@@ -54,5 +51,4 @@ def _build_payload(self, root_dir: Path) -> List[CoverageFilePayload]:
         return coverage_data
 
     def build_payload(self, root_dir: Path) -> Dict[str, List[CoverageFilePayload]]:
-        """Generate a CI Visibility coverage payload in JSON format"""
```
[the remark this answers is not in the document]
Comment (romainkomorndatadog) [reply ddtrace/internal/ci_visibility/api/_coverage_data.py:57]: It brought no value and was incorrect, to boot.

File: ddtrace/ext/test_visibility/coverage_lines.py [hunk 7]
```diff
@@ -1,3 +1,6 @@
+"""
+NOTE: BETA - this API is currently in development and is subject to change.
+"""
```
[the remark this answers is not in the document]
Comment (romainkomorndatadog) [reply ddtrace/ext/test_visibility/coverage_lines.py:3]: I don't think the `internal` vs `external` distinction should be used for release status, but more importantly, it's being used by the external portion of this API, so it should also be external.

File: tests/ci_visibility/api/fake_runner_all_fail.py (hunk 3 above)
Comment (romainkomorndatadog) [reply tests/ci_visibility/api/fake_runner_all_fail.py:8]: Same as for the other comment. Maybe in another PR.

**Comment by romainkomorndatadog** [comment]
@mabdinur , @ZStriker19 , could I get a review for the `trace_handlerts.py` file?

@tabgok , can I get a review on the... non-IDM IDM things?

**Review by tabgok — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**